crates/compiler/test_mono/generated/encode_derived_nested_record_string.txt

```diff
@@ -1,36 +1,44 @@
 procedure #Derived.0 (#Derived.1):
-    let #Derived_gen.1 : {Str} = Struct {#Derived.1};
-    let #Derived_gen.0 : {Str} = CallByName Encode.22 #Derived_gen.1;
+    let #Derived_gen.2 : {Str} = Struct {#Derived.1};
+    let #Derived_gen.1 : Str = StructAtIndex 0 #Derived_gen.2;
+    inc #Derived_gen.1;
+    dec #Derived_gen.2;
```

Comment:
Are we able to optimise away these refcounting ops?

Reply:
Yeah. We can avoid generating them entirely by not doing the wrapping/unwrapping done here. Currently doing so requires a symbol rename, which is expensive in the compiler (requires traversing and rebuilding the rest of the block statements). I plan to look into that more.

Reply:
Ok interesting, thanks!

Reply:
Got rid of these in 1d4a1e7040715c198acded7152ebf8f61f7f2dcd.